pyro/infer/mcmc/nuts.py

```diff
@@ -81,10 +81,10 @@ def __init__(self, model, step_size=None, adapt_step_size=False, transforms=None
         self._max_sliced_energy = 1000
 
     def _is_turning(self, z_left, r_left, z_right, r_right):
-        z_left = torch.stack([z_left[name] for name in self._r_dist])
-        r_left = torch.stack([r_left[name] for name in self._r_dist])
-        z_right = torch.stack([z_right[name] for name in self._r_dist])
-        r_right = torch.stack([r_right[name] for name in self._r_dist])
+        z_left = torch.cat([z_left[name] for name in self._r_dist])
```

Comment:
Same here. Is there a way to flatten this so that we do not need to have this rather stringent requirement of latent site shapes matching up?

Reply:
I wonder it too. ^^

Reply:
I was going to mention in your GP pr that this might better be done in a loop of accumulators, something like:
```py
diff_left = 0
diff_right = 0
for name in self._r_dist:
    dz = z_right[name] - z_left[name]
    diff_left += (dz * r_left[name]).sum()
    diff_right += (dz * r_right[name]).sum()
return diff_left < 0 or diff_right < 0
```